hal/src/nRF52840/ble_hal.cpp

```diff
@@ -1414,12 +1422,12 @@ int BleObject::Observer::stopScanning() {
 
 ble_gap_scan_params_t BleObject::Observer::toPlatformScanParams() const {
     ble_gap_scan_params_t params = {};
-    params.extended = 0;
+    params.extended = scanUseCoded_ ? 1 : 0;
     params.active = scanParams_.active;
     params.interval = scanParams_.interval;
```

Comment:
As per the API manual:
> If scan_phys contains both ref BLE_GAP_PHY_1MBPS and ref BLE_GAP_PHY_CODED interval shall be larger than or equal to twice the scan window.

Reply:
Could add a safety check here, but not sure how to communicate that back to the programmer.

Reply:
I think the programmer cares more about scanning interval. We can probably tunes the scanning window internally to meet the requirement only if both 1MBPS and coded PHY are used.

Reply:
Done, automatic limit of window to interval/2.  Modified the test app to demonstrate this.